core/src/main/scala/cats/Inject.scala

```diff
@@ -0,0 +1,47 @@
+package cats
+
+import cats.arrow.FunctionK
+import cats.data.Coproduct
+import cats.syntax.either._
+
+/**
+ * Inject type class as described in "Data types a la carte" (Swierstra 2008).
+ *
+ * @see [[http://www.staff.science.uu.nl/~swier004/publications/2008-jfp.pdf]]
+ */
+sealed abstract class Inject[F[_], G[_]] {
+  def inj: FunctionK[F, G]
+
+  def prj: FunctionK[G, λ[α => Option[F[α]]]]
```

Comment:
I’m fine with either. The current ones match the paper and Scalaz. Also, Matryoshka defines `project`, so I don’t mind the lack of collision 😁